specs/sharding/beacon-chain.md

```diff
@@ -423,19 +508,9 @@ def get_shard_proposer_index(beacon_state: BeaconState, slot: Slot, shard: Shard
     Return the proposer's index of shard block at ``slot``.
     """
     epoch = compute_epoch_at_slot(slot)
-    committee = get_shard_committee(beacon_state, epoch, shard)
-    seed = hash(get_seed(beacon_state, epoch, DOMAIN_SHARD_PROPOSER) + uint_to_bytes(slot))
-
-    # Proposer must have sufficient balance to pay for worst case fee burn
-    EFFECTIVE_BALANCE_MAX_DOWNWARD_DEVIATION = (
-        EFFECTIVE_BALANCE_INCREMENT - EFFECTIVE_BALANCE_INCREMENT
-        * HYSTERESIS_DOWNWARD_MULTIPLIER // HYSTERESIS_QUOTIENT
-    )
-    min_effective_balance = (
-        beacon_state.shard_gasprice * MAX_SAMPLES_PER_BLOCK // TARGET_SAMPLES_PER_BLOCK
-        + EFFECTIVE_BALANCE_MAX_DOWNWARD_DEVIATION
-    )
-    return compute_proposer_index(beacon_state, committee, seed, min_effective_balance)
+    seed = hash(get_seed(beacon_state, epoch, DOMAIN_SHARD_BLOB) + uint_to_bytes(slot) + uint_to_bytes(shard))
+    indices = get_active_validator_indices(state, epoch)
+    return compute_proposer_index(beacon_state, indices, seed)
```

Comment:
Especially if the data-tx topic is lightweight and global, as proposed now, it seems fine to me.

If we want different indices, we could use multiple index-wise shuffles for different list indices but same seed, and implement an alternative for the effective-balance bias that is currently applied to proposer computation. Or repeat the computation with a different seed if the proposer is already assigned to a shard (reuses existing code, but determining a role of a single validator then depends on others). I think I prefer the current status, and with mainnet validator counts it will be very rare anyway.